src/components/Card/Card.js

```diff
@@ -0,0 +1,88 @@
+import React, { useReducer } from 'react'
+import { useCart } from "../../context/CartContext"
+import { useAuth } from "../../context/AuthenticationContext"
+import { Link, useNavigate } from "react-router-dom";
+import { addItemInWishlist, updateCart, removeItemFromWishlist } from '../../utilities/helpers/http-helper';
+
+const Card = (props) => {
+
+    const { authState: { token, isLogin } } = useAuth();
```

Comment:
same goes here as well. if you destructure in the context itself, it will also make the code cleaner here and this must be used in multiple components, so why repeat the destructuring everywhere, just destructure at the source